### PR TITLE
removed refenreces to ICompletableFuture

### PR DESCRIPTION
**Pull request by alex-dukhno**
This PR is part of the effort of #15796 to remove references to ICompletableFuture.

@vbekiaris could you please take a look?

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/RingbufferAddAllCodec.java [hunk 1]
```diff
@@ -104,7 +101,7 @@ public static ClientMessage encodeRequest(java.lang.String name, java.util.Colle
     public static class ResponseParameters {
 
         /**
-         * the ICompletableFuture to synchronize on completion.
+         * the CompletionStage to synchronize on completion.
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/RingbufferAddAllCodec.java]: These codecs are auto-generated and we dont allow them to be changed by hand. You need to update [this](https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/Ringbuffer.yaml#L250) protocol definition, send a pull request to protocol repo and regenerate the codecs as described in the README. (`python3 generator.py -r /your/hazelcast/repo/path`)

Comment (alex-dukhno) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/RingbufferAddAllCodec.java]: I submitted the [PR](https://github.com/hazelcast/hazelcast-client-protocol/pull/253) and pushed generated codec.
Thanks for help.

**Comment by mdumandag** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/cp/IAtomicReference.java [hunk 2]
```diff
@@ -27,11 +27,11 @@
  * to the {@link java.util.concurrent.atomic.AtomicReference}.
  * <p>
  * Asynchronous variants have been introduced in version 3.7.
- * Async methods immediately return an {@link ICompletableFuture} from which
+ * Async methods immediately return an {@link CompletionStage} from which
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/cp/IAtomicReference.java]: minor: "an CompletionStage" -> "a CompletionStage"

File: hazelcast/src/main/java/com/hazelcast/cp/IAtomicReference.java [hunk 3]
```diff
@@ -27,11 +27,11 @@
...
  * the operation's result can be obtained either in a blocking manner or by
  * registering a callback to be executed upon completion. For example:
  * <pre>
- * ICompletableFuture&lt;E&gt; future = atomicRef.getAsync();
+ * CompletionStage&lt;E&gt; future = atomicRef.getAsync();
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/cp/IAtomicReference.java]: The following example must be also updated. The replacement for `ICompletableFuture.andThen` is `CompletionStage.whenCompleteAsync`, eg see https://github.com/hazelcast/hazelcast/blob/83d814eefb9864187b3a43736d29422a1581f0aa/hazelcast/src/main/java/com/hazelcast/map/IMap.java#L650-L657

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
@mdumandag should we first merge the client-protocol PR before merging this?

**Comment by mdumandag** [comment]
@vbekiaris We generally merge the Hazelcast side first and then the protocol. Since both of them are approved and tests are passed, I am merging them.

**Pull request merged**